api/test/round-tracker.test.js

```diff
@@ -435,60 +435,68 @@ describe('Round Tracker', () => {
           ]
         )
       })
-      it('scales the task count to reach desired tasks executed', async () => {
-        const prevSparkRoundNumber = await mapCurrentMeridianRoundToSparkRound({
-          meridianContractAddress: '0x1a',
-          meridianRoundIndex: 120n,
-          roundStartEpoch: 321n,
-          pgClient,
-          recordTelemetry: createTelemetryRecorderStub().recordTelemetry
-        })
-        await pgClient.query(
-          'UPDATE spark_rounds SET measurement_count = $1 WHERE id = $2',
-          [TASKS_EXECUTED_PER_ROUND / 2, prevSparkRoundNumber]
-        )
-        // It should double the node task count as the previous round only
-        // produces half of the desired measurements
-        const { recordTelemetry, telemetry } = createTelemetryRecorderStub()
-        const sparkRoundNumber = await mapCurrentMeridianRoundToSparkRound({
-          meridianContractAddress: '0x1a',
-          meridianRoundIndex: 121n,
-          roundStartEpoch: 321n,
-          pgClient,
-          recordTelemetry
+
+      for (const measurementsFactor of [2, 0.5]) {
+        it(`scales ${measurementsFactor > 1 ? 'down' : 'up'} the task count to reach desired tasks executed`, async () => {
+          const prevSparkRoundNumber = await mapCurrentMeridianRoundToSparkRound({
+            meridianContractAddress: '0x1a',
+            meridianRoundIndex: 120n,
+            roundStartEpoch: 321n,
+            pgClient,
+            recordTelemetry: createTelemetryRecorderStub().recordTelemetry
+          })
+          await pgClient.query(
+            'UPDATE spark_rounds SET measurement_count = $1 WHERE id = $2',
+            [TASKS_EXECUTED_PER_ROUND * measurementsFactor, prevSparkRoundNumber]
+          )
+          const { recordTelemetry, telemetry } = createTelemetryRecorderStub()
+          const sparkRoundNumber = await mapCurrentMeridianRoundToSparkRound({
+            meridianContractAddress: '0x1a',
+            meridianRoundIndex: 121n,
+            roundStartEpoch: 321n,
+            pgClient,
+            recordTelemetry
+          })
+          const { rows: [sparkRound] } = await pgClient.query(
+            'SELECT * FROM spark_rounds WHERE id = $1',
+            [sparkRoundNumber]
+          )
+          assert.strictEqual(sparkRound.max_tasks_per_node, Math.floor(BASELINE_TASKS_PER_NODE / measurementsFactor))
+          const { rows: retrievalTasks } = await pgClient.query(
+            'SELECT * FROM retrieval_tasks WHERE round_id = $1',
+            [sparkRoundNumber]
+          )
+          const expectedRetrievalTasksCount = Math.floor(
+            Number(BigInt(BASELINE_TASKS_PER_NODE) *
+            BigInt(TASKS_EXECUTED_PER_ROUND) /
+            BigInt(TASKS_EXECUTED_PER_ROUND * measurementsFactor)) *
+            ROUND_TASKS_TO_NODE_TASKS_RATIO
+          )
```

Comment:
I found it difficult to spot what is being converted to `Number`.

Proposed formatting:

```suggestion
          const expectedRetrievalTasksCount = Math.floor(
            Number(
              BigInt(BASELINE_TASKS_PER_NODE) *
              BigInt(TASKS_EXECUTED_PER_ROUND) /
              BigInt(TASKS_EXECUTED_PER_ROUND * measurementsFactor)
            ) *
            ROUND_TASKS_TO_NODE_TASKS_RATIO
          )
```

Or alternatively

```suggestion
          const expectedRetrievalTasksCount = Math.floor(
            Number(
              BigInt(BASELINE_TASKS_PER_NODE) *
              BigInt(TASKS_EXECUTED_PER_ROUND) /
              BigInt(TASKS_EXECUTED_PER_ROUND * measurementsFactor)
            ) * ROUND_TASKS_TO_NODE_TASKS_RATIO
          )
```

Reply:
nice!